### PR TITLE
G3 5 deploy geneweaver api to shared clusters

### PR DESCRIPTION
**Pull request by bergsalex**
This PR adds:
- Deployments Definitions for Github
- Kubernetes Deployment Yaml
- A Dockerfile
- A Skaffold definition file

It also cleans up the configuration/settings relating to database config.

### CODE REVIEW

**Review by gerring — COMMENTED** [review]
Possibly the most confusing yaml I have ever looked at, bewildered.

**Comment by bergsalex** [comment]
> Possibly the most confusing yaml I have ever looked at, bewildered.

@gerring I added some readme content to help explain it.

**Review by desais-jax — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM python:3.9
```
Comment (gerring) [inline Dockerfile:1]: 3.9?

File: src/geneweaver/api/core/config_class.py [hunk 2]
```diff
@@ -38,7 +42,7 @@ def assemble_db_connection(
         "openid profile email": "read",
     }
     JWT_PERMISSION_PREFIX: str = "approle"
-    AUTH_CLIENT_ID: str = "oVm9omUtLBpVyL7YfJA8gp3hHaHwyVt8"
+    AUTH_CLIENT_ID: str = "T7bj6wlmtVcAN2O6kzDRwPVFyIj4UQNs"
```
Comment (gerring) [inline src/geneweaver/api/core/config_class.py:45]: I assume this is not a hard coded cred. There is no way that much yaml and then still have things like this if so.

Comment (bergsalex) [reply src/geneweaver/api/core/config_class.py:45]: It's the dev client ID. It's not a secret and makes sense to put in ad a default. 

The specific line change here relates to what made sense to put into the kubernetes config maps.

File: Dockerfile (hunk 1 above)
Comment (bergsalex) [reply Dockerfile:1]: We should be good on 3.10 (or even 3.11), we're already running tests on both. We should also be okay using the slim variants. 

I decided those changes should go into future PRs, but happy to make them here if you think it's needed here.

**Review by gerring — APPROVED** [review]

**Pull request merged**